tests/EStimLibrary.UnitTests/Core/HardwareInterfaces/NeuralInterfaceManagerTests.cs

```diff
@@ -0,0 +1,207 @@
+        Assert.Equal(interfaceType, neuralInterface.GetType());
+
+        // That the new neural interface object is created with the given object[] parameters, in this case, none.
+        if (interfaceType == typeof(GelPad) && neuralInterface is GelPad gelPad) {
+            Assert.Equal(new GelPad().NumContacts, gelPad.NumContacts); 
+        }
+    }
+
+    /// <summary>
+    /// The manager assigns global contact IDs for the new neural interface. 
+    /// Test that this ID is returned by the method CreateAndRegisterNeuralInterface.
+    /// </summary>
+    /// <param name="interfaceType">The type of the hardware interface, for example: ContactGroup, GelPad</param>
+    /// <param name="interfaceSpecificParams">Parameters specific to that interface</param>
+    [Theory]
+    [InlineData(typeof(ContactGroup), new object[]{1})]
+    [InlineData(typeof(GelPad), new object[]{})]
+    public void CreateAndRegisterNeuralInterface_ShouldReturnAssignedGlobalContactIDs(Type interfaceType, object[] interfaceSpecificParams)
+    {
+        // Arrange
+        var NIManager = new NeuralInterfaceManager();
+
+        // Act
+        var contactIds = NIManager.CreateAndRegisterNeuralInterface(interfaceType, interfaceSpecificParams, out _);
+
+        // Assert
+        Assert.NotEmpty(contactIds);
+        Assert.All(contactIds, id => Assert.True(NIManager.IsValidContactId(id)));
+    }
+
+    /// <summary>
+    /// The manager assigns global contact IDs for the new neural interface. 
+    /// Test that this contact assignment upholds global contact ID uniqueness when the interface is the first one added.
+    /// </summary>
+    /// <param name="interfaceType">The type of the hardware interface, for example: ContactGroup, GelPad</param>
+    /// <param name="interfaceSpecificParams">Parameters specific to that interface</param>
+    [Theory]
+    [InlineData(typeof(ContactGroup), new object[]{1})]
+    [InlineData(typeof(GelPad), new object[]{})]
+    public void CreateAndRegisterNeuralInterface_ShouldAssignUniqueGlobalContactIDs_WhenFirstInterface(Type interfaceType, object[] interfaceSpecificParams)
+    {
+        // Arrange
+        var NIManager = new NeuralInterfaceManager();
+
+        // Act
+        var contactIds = NIManager.CreateAndRegisterNeuralInterface(interfaceType, interfaceSpecificParams, out _);
+
+        // Assert
+        Assert.Equal(contactIds, contactIds.Distinct());
+    }
+
+    /// <summary>
+    /// The manager assigns global contact IDs for the new neural interface. 
+    /// Test that this contact assignment upholds global contact ID uniqueness when the interface is not the first one added.
+    /// </summary>
+    /// <param name="interfaceType">The type of the hardware interface, for example: ContactGroup, GelPad</param>
+    /// <param name="interfaceSpecificParams">Parameters specific to that interface</param>
+    [Theory]
+    [InlineData(typeof(ContactGroup), new object[]{1})]
```

Comment:
To make passing the same data into all of these methods easier, you can use the MemberData struct instead and store all the options in a single struct which you then list before each method with the [MemberData(...)] tag

Reply:
Another data case to add to this MemberData item is a ContactGroup with a little more than one contact, and one with a lot more than one contact (test the integer limits!)